### PR TITLE
Set initial values of SetupOptions parameters from GlobalSetup

### PR DESCRIPTION
**Pull request by mrguilima**
These calls are useful for interactive mode of the UImessenger interface, which Celeritas is currently not using.

### CODE REVIEW

**Review by sethrj — COMMENTED** [review]
Since `DefaultValue` didn't work, can you set the actual default values of the properties in the `GlobalSetup()` constructor?

File: app/demo-geant-integration/GlobalSetup.cc [hunk 1]
```diff
@@ -43,38 +43,51 @@ GlobalSetup::GlobalSetup()
...
+        if (celeritas::Device::num_devices() > 0)
+        {
+            cmd.SetDefaultValue("524288");
+            options_->max_num_events = 524288;
+        }
+        else
+        {
+            cmd.SetDefaultValue("64");
+            options_->max_num_events = 64;
+        }
```
Comment (sethrj) [inline app/demo-geant-integration/GlobalSetup.cc:55]: I think it's still ok to remove `SetDefaultValue`! I just wanted it to be replaced with *setting* the default value like the original code intended. If we still do want to set the default, maybe we could use `to_string` to simplify the code and avoid duplicating the values?
```suggestion
        options_->max_num_events = celeritas::Device::num_devices() > 0 ? 524288 : 64;
        cmd.SetDefaultValue(std::to_string(options_->max_num_events));
```

Comment (mrguilima) [reply app/demo-geant-integration/GlobalSetup.cc]: Now that the default values are being set here as well, I thought there is some (documentation) value in keeping the calls to SetDefaultValue(). I keep the to_string() option for now.

**Comment by sethrj** [comment]
@mrguilima You'll need to merge master since `heap_size` and `stack_size` have had to move to `SetupOptions`.

File: app/demo-geant-integration/GlobalSetup.cc (hunk 1 above)
Comment (sethrj) [reply app/demo-geant-integration/GlobalSetup.cc]: Perfect!

**Review by sethrj — APPROVED** [review]

**Pull request merged**